### PR TITLE
Feature/alcs 1446 inquiry advanced search

### PR DESCRIPTION
**Pull request by mhuseinov**
- inquiry advanced search
- fix typo in inquiry board code
- bonus: slight increase of search API coverage

### CODE REVIEW

File: alcs-frontend/src/app/features/search/inquiry-search-table/inquiry-search-table.component.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        <app-application-type-pill *ngIf="!element.open" [type]="CLOSED_TYPE" />
+      </td>
+    </ng-container>
+
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr (click)="onSelectRecord(row)" mat-row *matRowDef="let row; columns: displayedColumns"></tr>
+
+    <tr class="mat-row no-data" *matNoDataRow>
+      <td class="text-center" colspan="4">
+        <div><b>No non-applications found.</b></div>
```
Comment (dhaselhan) [inline alcs-frontend/src/app/features/search/inquiry-search-table/inquiry-search-table.component.html:61]: No inquiries found?

File: alcs-frontend/src/app/features/search/inquiry-search-table/inquiry-search-table.component.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    (page)="onPageChange($event)"
+    aria-label="Select page"
+  ></mat-paginator>
+</ng-container>
+<div *ngIf="isLoading" class="center">
+  <mat-spinner />
+</div>
+
+<div class="no-data" *ngIf="totalCount === 0">
+  <div><b>No non-applications found.</b></div>
```
Comment (dhaselhan) [inline alcs-frontend/src/app/features/search/inquiry-search-table/inquiry-search-table.component.html:80]: Inquiries

File: services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    query = query
+      .orderBy(
+        sortQuery,
+        searchDto.sortDirection,
+        searchDto.sortDirection === 'ASC' ? 'NULLS FIRST' : 'NULLS LAST',
+      )
+      .offset((searchDto.page - 1) * searchDto.pageSize)
+      .limit(searchDto.pageSize);
+
+    // console.log(query.getSql());
```
Comment (dhaselhan) [inline services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts:41]: delete

File: services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+      case 'status':
+        return '"inquirySearch"."open"';
+
+      default:
+      case 'dateSubmitted':
+        return '"inquirySearch"."date_submitted_to_alc"';
+    }
+  }
+
+  private compileGroupBySearchQuery(
```
Comment (dhaselhan) [inline services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts:74]: Do you need this?

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

File: alcs-frontend/src/app/features/search/inquiry-search-table/inquiry-search-table.component.html (hunk 1 above)
Comment (mhuseinov) [reply alcs-frontend/src/app/features/search/inquiry-search-table/inquiry-search-table.component.html]: done

File: alcs-frontend/src/app/features/search/inquiry-search-table/inquiry-search-table.component.html (hunk 2 above)
Comment (mhuseinov) [reply alcs-frontend/src/app/features/search/inquiry-search-table/inquiry-search-table.component.html]: done

File: services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts (hunk 3 above)
Comment (mhuseinov) [reply services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts]: done

**Pull request merged**